connexion/decorators/decorator.py

```diff
@@ -38,8 +39,22 @@ def __call__(self, function):
         :rtype: types.FunctionType
         """
         if has_coroutine(function, self.api):
-            from .coroutine_wrappers import get_request_life_cycle_wrapper
-            wrapper = get_request_life_cycle_wrapper(function, self.api, self.mimetype)
+            @functools.wraps(function)
+            async def wrapper(*args, **kwargs):
+                connexion_request = self.api.get_request(*args, **kwargs)
+                while asyncio.iscoroutine(connexion_request):
```

Comment:
FYI: There is also `inspect.iscoroutine()` for checking async functions, both Quart and Starlette seem to be using both. But as long as there is no more `yield from`, I believe they should be similar

Reply:
Good to know!

From the docs:
> This method is different from inspect.iscoroutine() because it returns True for generator-based coroutines.

We still need this if we want to support user code that uses generator-based coroutine view functions. While support for generator-based coroutines in Python is deprecated, it is only scheduled for removal in Python 3.10. Since they're still supported by Python, I think we should support them too, and this way we don't break existing applications.

Reply:
Yes, agreed.
I guess it is just something to keep in mind, as I ran into it when connexion considered something async but the framework (Starlette) didn't.